File: FrontEnd/frontend/src/services/productoCommandService.ts
import axios from 'axios';
import {type CrearProductoCommand} from '../types/CrearProductoCommand';

const API_URL = 'http://localhost:8080/api/productos/commands';

export const crearProducto = async (producto: CrearProductoCommand): Promise<number> => {
  const response = await axios.post<number>(API_URL, producto);
  return response.data;
};

export const actualizarProducto = async (id: number, producto: CrearProductoCommand): Promise<void> => {
  await axios.put(`${API_URL}/${id}`, producto);
};

export const eliminarProducto = async (id: number): Promise<void> => {
  await axios.delete(`${API_URL}/${id}`);
};

export const restaurarProducto = async (id: number): Promise<void> => {
  await axios.put(`${API_URL}/restaurar/${id}`);
};

File: BackEnd/patroncqrs/src/main/java/com/cqrs/patroncqrs/command/service/ProductoCommandService.java
package com.cqrs.patroncqrs.command.service;

import com.cqrs.patroncqrs.command.dto.CrearProductoCommand;

public interface ProductoCommandService {
    Long crearProducto(CrearProductoCommand command);
    void actualizarProducto(Long id, CrearProductoCommand command);
    void eliminarProducto(Long id);
    void restaurarProducto(Long id);
}
